cli/src/commands/migration/status.test.ts

```diff
@@ -0,0 +1,256 @@
+                },
+                xata_version: {
+                  name: '_version',
+                  type: 'integer',
+                  default: '0',
+                  nullable: false,
+                  unique: false,
+                  comment: ''
+                }
+              },
+              indexes: {},
+              primaryKey: ['xata_id'],
+              foreignKeys: null,
+              checkConstraints: null,
+              uniqueConstraints: null
+            }
+          }
+        }
+      })
+    };
+  }
+
+  throw new Error(`Unexpected fetch request: ${url} ${request.method}`);
+};
+
+export const fetchEmptyStatus = (url: string, request: any) => {
+  if (url === `${baseUrl}/migrations/status` && request.method === 'GET') {
+    return {
+      ok: true,
+      status: 204,
+      json: async () => null // 204, no content
+    };
+  } else {
+    return baseFetch(url, request);
+  }
+};
+
+// Note: the following mocks follow the fetch<type>_<status>Status pattern
+
+const fetchComplete_CompletedStatus = (url: string, request: any) => {
+  if (url === `${baseUrl}/migrations/status` && request.method === 'GET') {
+    return {
+      ok: true,
+      json: async () => {
+        return {
+          completedAt: '2024-07-22T09:41:54.695893Z',
+          jobID: 'mig_job_cqf2inp20majlmbjukh0',
+          status: 'completed',
+          type: 'complete'
+        };
+      }
+    };
+  } else {
+    return baseFetch(url, request);
+  }
+};
+
+export const fetchStart_CompletedStatus = (url: string, request: any) => {
+  if (url === `${baseUrl}/migrations/status` && request.method === 'GET') {
+    return {
+      ok: true,
+      json: async () => {
+        return {
+          completedAt: '2024-07-22T16:52:55.125163Z',
+          description: [
+            {
+              update: {
+                mapping: {
+                  description: '_pgroll_new_description'
+                },
+                name: 'description',
+                table: 'table',
+                type: 'column'
+              }
+            }
+          ],
+          jobID: 'mig_job_cqf8spc40ertajao726g',
+          status: 'completed',
+          type: 'start'
+        };
+      }
+    };
+  } else {
+    return baseFetch(url, request);
+  }
+};
+
+const fetchRollback_CompletedStatus = (url: string, request: any) => {
+  if (url === `${baseUrl}/migrations/status` && request.method === 'GET') {
+    return {
+      ok: true,
+      json: async () => {
+        return {
+          completedAt: '2024-07-22T16:59:46.41859Z',
+          jobID: 'mig_job_cqf8vvk40ertajao7270',
+          status: 'completed',
+          type: 'rollback'
+        };
+      }
+    };
+  } else {
+    return baseFetch(url, request);
+  }
+};
+
+promptsMock.mockReturnValue({ confirm: true, database: 'db1', workspace: 'test-1234' });
+
+describe('migration status', () => {
+  test('correctly detects if migration status is run in a project with no migrations', async () => {
```

Comment:
Should we add a failed case? And maybe a test for non pgroll branches?

Reply:
Good idea, on it